minsar/utils/download_asfserial_rsmas.py

```diff
@@ -67,15 +88,64 @@ def main(iargs=None):
     return None
 
 
-def generate_files_csv(slc_dir, custom_template_file):
+def generate_files_csv(slc_dir, custom_template_file, tupleParam):
     """ Generates a csv file of the files to download serially.
     Uses the `awk` command to generate a csv file containing the data files to be download
     serially. The output csv file is then sent through the `sed` command to remove the first five
     empty values to eliminate errors in download_ASF_serial.py.
     """
-
-    dataset_template = Template(custom_template_file)
-    dataset_template.options.update(PathFind.correct_for_ssara_date_format(dataset_template.options))
+    
+    inps = tupleParam[0]
+    dataset_template = tupleParam[1]
+    if inps.seasonalStartDate is not None and inps.seasonalEndDate is not None:
```

Comment:
Please comment this. I have no idea what's happening here, especially with the huge nested conditional blocks.

Reply:
```
# provides necessary standard data to run the method
    inps = tuple_inputs[0]
    dataset_template = tuple_inputs[1]
    # in the case seasonal dates are provided, more data is required for manipulations to provide intended result
    if inps.seasonalStartDate is not None and inps.seasonalEndDate is not None:
```